prov/efa/src/efa_cq.c

```diff
@@ -331,11 +359,210 @@ int efa_cq_open(struct fid_domain *domain_fid, struct fi_cq_attr *attr,
 err_destroy_pool:
 	ofi_bufpool_destroy(cq->wce_pool);
 err_destroy_cq:
-	ibv_destroy_cq(cq->ibv_cq);
+	ibv_destroy_cq(ibv_cq_ex_to_cq(cq->ibv_cq_ex));
 err_free_util_cq:
 	ofi_cq_cleanup(&cq->util_cq);
 err_free_cq:
 	free(cq);
 	return ret;
 }
 
+
+#if EFA_UNIT_TEST
+
+#include "efa_unit_tests.h"
+#include "efa_unit_test_utils.h"
+
+struct efa_cq_readfrom_state {
+	struct efa_cq *efa_cq;
```

Comment:
Can we put `efa_cq` to `efa_unit_test_resource`?

Reply:
Same as above. Avoid hard dependency on efa device.